File: Difficulty/Easy/[205] Isomorphic Strings/solution.cpp
/*
 * @lc app=leetcode id=205 lang=cpp
 *
 * [205] Isomorphic Strings
 */

// @lc code=start
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char, char> um1, um2;
        for (int i = 0; i < s.size(); i++) {
            if (um1.find(t[i]) != um1.end()) {
                if (s[i] != um1[t[i]])
                    return false;
            }else
                um1[t[i]] = s[i];
            if (um2.find(s[i]) != um2.end()) {
                if (t[i] != um2[s[i]])
                    return false;
            }else
                um2[s[i]] = t[i];
        }
        return true;
    }
};
// @lc code=end

File: Difficulty/Easy/[205] Isomorphic Strings/solution.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        d1, d2 = dict(), dict()
        for i in range(len(s)):
            if d1.get(t[i], 0):
                if s[i] != d1[t[i]]:
                    return False
            else:
                d1[t[i]] = s[i]
            if d2.get(s[i], 0):
                if t[i] != d2[s[i]]:
                    return False
            else:
                d2[s[i]] = t[i]
        return True
